core/units.c

```diff
@@ -183,3 +186,49 @@ const struct units *get_units()
 {
 	return &prefs.units;
 }
+
+int parse_duration(const char *buffer, duration_t *time, bool dot_as_seperator)
+{
+	char *line = (char*)buffer;
```

Comment:
I don't like that const cast, but understand its raison d'être: the `strtol` interface.

If you want to get rid of it, you could use a helper function such as
```
#include <charconv>

template <typename T>
static const char *parse_int(const char *buf, size_t size, T &i)
{
    return std::from_chars(buf, buf + size, i).ptr;
}
```
or directly write below
```
line = std::from_chars(line, line + 10, values[n++]);
```
But from my side, you can also leave it as as.